drivers/dma/dma_stm32.h

```diff
@@ -7,6 +7,11 @@
 #ifndef DMA_STM32_H_
 #define DMA_STM32_H_
 
+#include <soc.h>
+#include <drivers/dma.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/stm32_clock_control.h>
```

Comment:
Do we need all these ?
At least in this commit, this doesn't seem obvious.

Reply:
"include and forget", so probably not, the thing that was needed were the include(s) for `struct stm32_pclken` because that is used in the header file, and the idea was that zephyr includes must be self contained, right ?

Let me figure out which one is really needed and which ones aren't.

Reply:
I cleaned up the #includes, hope it is OK now?